tessera-core/src/main/java/com/quorum/tessera/node/PartyInfoStore.java

```diff
@@ -39,6 +40,12 @@ public PartyInfoStore(final ConfigService configService) {
     public synchronized void store(final PartyInfo newInfo) {
         recipients.addAll(newInfo.getRecipients());
         parties.addAll(newInfo.getParties());
+
+        //update the sender to have been seen recently
+        final Party sender = new Party(newInfo.getUrl());
```

Comment:
There's a small issue here. If the url has been added to the config without a trailing '/' (like in 7node examples) then we end up with duplicate entries, e.g.:
'''
&lt;snip&gt;
"peers":[
        {"url":"http://localhost:9007","lastContact":null},
        {"url":"http://localhost:9005","lastContact":null},
        {"url":"http://localhost:9006","lastContact":null},
        {"url":"http://localhost:9003","lastContact":null},
        {"url":"http://localhost:9004","lastContact":null},
        {"url":"http://localhost:9001","lastContact":null},
        {"url":"http://localhost:9002","lastContact":null},
        {"url":"http://localhost:9006/","lastContact":"2019-01-03T03:13:38.984Z"},
        {"url":"http://localhost:9005/","lastContact":"2019-01-03T03:13:38.984Z"},
        {"url":"http://localhost:9004/","lastContact":"2019-01-03T03:13:38.985Z"},
        {"url":"http://localhost:9003/","lastContact":"2019-01-03T03:13:38.986Z"},
        {"url":"http://localhost:9002/","lastContact":"2019-01-03T03:13:38.986Z"},
        {"url":"http://localhost:9001/","lastContact":"2019-01-03T03:13:38.982Z"},
        {"url":"http://localhost:9007/","lastContact":"2019-01-03T03:13:38.987Z"}
],
&lt;snip&gt;
'''
I think the root cause of this might be in the constructor for PartyInfoServiceImpl, where the initialParties value is taken from the configured peers.
I also notice that we have a bit of duplicate code dotted around to add the trailing backslash. Maybe it could all be consolidated into PartyInfoStore?

Reply:
BTW after posting my comment, I pulled the latest 7nodes example and noticed that has been changed to include the forward-slash on the config.
However, we should still modify the Tessera code so that with/without forward-slash are treated the same.

Reply:
I will filter out the URLs without a /, as these will always be null.